### ISSUE TITLE
C-x C-s does not save notebooks

### ISSUE THREAD

**Issue by sam-s** [millejoh/emacs-ipython-notebook#130]
`C-x C-s` (`save-some-buffers`) does not offer to save notebooks (this is, IMO, _bad_).
However, when I try to exit Emacs, I am told about unsaved notebooks (this is _good_, but inconsistent with the above).
It would be much better if the notebooks were saved in a consistent manner.

**Comment by millejoh** [comment]
I am open to suggestions on this one. The reason this doesn't work they way it should is because notebook buffers are not files (i.e. `'buffer-file-name` is nil), but rather cleverly display JSON data structures that are passed to and from the Jupyter server.

Setting `buffer-file-name` is not a good idea - I found this out the hard way when I once accidentally overwrote the ipython file on disk when exiting emacs.

I tried setting `buffer-offer-save` to t, but then you would have to make sure to always call `(save-some-buffers nil t)` and emacs prompts for a file name to save the file as.

I need a way to reliably ensure that `ein:notebook-save-notebook-command` is called whenever `save-buffer` might be called on the notebook buffer. Any ideas?
